WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -143,6 +143,9 @@ struct ItemListView: View {
                     searchTerm: $searchTerm
                 ) { _ in
                     itemListContent(selectedItemListType)
+                        .onDisappear() {
+                            searchItemsController.clearSearchItems(baseItem: .root)
+                        }
```

Comment:
Problem is, this clears it too often.

For example:
1. Search for a variable parent product, 
2. Tap the parent product to open it
3. Tap back – you'll see that there's no search results and an infinite loading indicator.

The same thing happens if you check out while viewing search then come back to edit the cart.

Reply:
Thanks for addressing this. You're right. I tested. The search is only cleared when it needs to be cleared.